### ISSUE TITLE
aria-colindex flagged as not allowed on <div role="columnheader">

### ISSUE THREAD

**Issue by HadleyLuker** [validator/validator#1369]
Hello! I'm trying to figure out if I've run into a bug or if I am misunderstanding the specs surrounding `columnheader` and `aria-colindex`.

Given this code (simplified from a [MUI data grid](https://mui.com/x/react-data-grid/) seen in the wild):
```html
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>Test</title>
  </head>
  <body>
    <div role="grid">
      <div role="row">
        <div role="columnheader" aria-colindex="1">Header R1C1</div>
        <div role="columnheader" aria-colindex="2">Header R1C2</div>
        <div role="columnheader" aria-colindex="3">Header R1C3</div>
        <div role="columnheader" aria-colindex="4">Header R1C4</div>
        <div role="columnheader" aria-colindex="5">Header R1C5</div>
        <div role="columnheader" aria-colindex="6">Header R1C6</div>
        <div role="columnheader" aria-colindex="7">Header R1C7</div>
        <div role="columnheader" aria-colindex="8">Header R1C8</div>
      </div>
      <div role="row">
        <div role="cell" aria-colindex="1">Cell R2C1</div>
        <div role="cell" aria-colindex="2">Cell R2C2</div>
        <div role="cell" aria-colindex="3">Cell R2C3</div>
        <div role="cell" aria-colindex="4">Cell R2C4</div>
        <div role="cell" aria-colindex="5">Cell R2C5</div>
        <div role="cell" aria-colindex="6">Cell R2C6</div>
        <div role="cell" aria-colindex="7">Cell R2C7</div>
        <div role="cell" aria-colindex="8">Cell R2C8</div>
      </div>
    </div>
  </body>
</html>

```

The validator returns this error for each of the `<div role="columnheader">` elements.

> Attribute `aria-colindex` not allowed on element [`div`](https://html.spec.whatwg.org/multipage/#the-div-element) at this point.

However, I'm not sure why.
- [`aria-colindex` inherits into the `columnheader` role](https://www.w3.org/TR/wai-aria/#aria-colindex)
- Each value is an integer greater than or equal to 1, greater than the `aria-colindex` value of any previous elements within the same row, and less than or equal to the number of columns in the full table
- [The role of `columnheader` is permitted on `div`s](https://www.w3.org/TR/html-aria/#el-div)
- The role nesting seems permitted: [the `grid` role can own an element with a `row` role](https://www.w3.org/TR/wai-aria/#grid), and [the `row` role can own an element with a `columnheader` role](https://www.w3.org/TR/wai-aria/#row)

I initially wondered if it was because all columns are present (the ARIA spec says `aria-colindex` is not necessary in this case), but the errors persist even if you include only the rows with `aria-colindex="2"` through` aria-colindex="5"`, for example. It also throws the errors for Examples 15 and 16 from the ARIA 1.1 spec (in the [`aria-colindex` section](https://www.w3.org/TR/wai-aria-1.1/#aria-colindex)).

Is the validator mistakenly flagging this as an issue, or have I missed something that disallows this element/role/property combination?

**Comment by sideshowbarker** [comment]
@scottaohara @stevefaulkner is the checker behavior out of conformance with this spec on this?

**Comment by ilia-brykin** [comment]
@hadleyel You didn't miss anything, it's a bug in the validator. Here is an example from the official documentation.
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-colindex

Will you fix this?

**Comment by Ignacio-de-Loyola** [comment]
https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/#ex3_label

In APG Example 3, it was guided to insert aria-colindex and aria-rowindex into <tr>, but The validator returns this error for each of the <th aria-colindex="1">, <th aria-rowindex="2"> elements. it's a bug in the validator?

**Comment by sideshowbarker** [comment]
> [w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids#ex3_label](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/#ex3_label)
> 
> This is a similar problem to the above. In APG Example 3, it was guided to insert aria-colindex and aria-rowindex into `<tr>`, but The validator returns this error for each of the `<th aria-colindex="1">`, `<th aria-rowindex="2">` elements. it's a bug in the validator?

Please raise a new/separate issue for that.
